app/qml/settings/MMSettingsPage.qml

```diff
@@ -183,6 +186,40 @@ MMPage {
         onClicked: __appSettings.autolockPosition = !checked
       }
 
+      MMLine {}
+
+      MMSettingsComponents.MMSettingsSwitch {
+        width: parent.width
+        title: qsTr("Enable haptic feedback")
+        description: qsTr("While recording, using buttons will give haptic feedback")
+        checked: __appSettings.useHaptics
+
+        onClicked: {
+          __appSettings.useHaptics = !checked
+          if ( checked && __androidUtils.isAndroid ) {
+            permissionDialog.open()
+          }
+        }
+      }
+
+      MMSettingsComponents.MMSettingsSwitch {
+        width: parent.width
+        title: qsTr("Enable sound ")
+        checked: __appSettings.useHapticsSound
+        visible: __appSettings.useHaptics
+
+        onClicked: __appSettings.useHapticsSound = !checked
+      }
+
+      MMSettingsComponents.MMSettingsSwitch {
+        width: parent.width
+        title: qsTr("Enable vibration")
```

Comment:
How about a single switch with the following options: `No haptic feedback`, `Sound`, `Vibration`, `Sound & Vibration`?